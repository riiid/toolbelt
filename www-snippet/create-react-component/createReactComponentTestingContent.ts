import { createImportRaw } from "./createImportRaw.ts";
import { CreateReactComponentOption } from "./CreateReactComponentOption.ts";

export const createReactComponentTestingContent = (
  name: string,
  option: CreateReactComponentOption,
) => {
  return `import { render } from "@testing-library/react";
${
    createImportRaw(
      option.onlyNamedExport ? undefined : name,
      option.onlyNamedExport ? [name] : undefined,
      `./${name}`,
    )
  }

describe("<${name} />", () => {
  it("should render children correctly", () => {
    const content = "Lorem Ipsum";
    const { queryByText } = render(<${name}>{content}</${name}>);
    expect(queryByText("Lorem Ipsum")).toBeInTheDocument();
  });
});
`;
};
