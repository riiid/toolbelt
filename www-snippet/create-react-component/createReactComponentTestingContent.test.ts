import { createReactComponentTestingContent } from "./createReactComponentTestingContent.ts";
import { assertSnapshot } from "https://deno.land/std@0.138.0/testing/snapshot.ts";

Deno.test(
  `should return correct react component testing.`,
  async (t) => {
    await assertSnapshot(
      t,
      createReactComponentTestingContent("LoremIpsum", {}),
    );
    await assertSnapshot(
      t,
      createReactComponentTestingContent("LoremIpsum", {
        onlyNamedExport: true,
      }),
    );
  },
);
