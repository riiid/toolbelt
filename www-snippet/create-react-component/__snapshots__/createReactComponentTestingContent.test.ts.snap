export const snapshot = {};

snapshot[`should return correct react component testing. 1`] = `
'import { render } from "@testing-library/react";
import LoremIpsum from "./LoremIpsum";

describe("<LoremIpsum />", () => {
  it("should render children correctly", () => {
    const content = "Lorem Ipsum";
    const { queryByText } = render(<LoremIpsum>{content}</LoremIpsum>);
    expect(queryByText("Lorem Ipsum")).toBeInTheDocument();
  });
});
'
`;

snapshot[`should return correct react component testing. 2`] = `
'import { render } from "@testing-library/react";
import { LoremIpsum } from "./LoremIpsum";

describe("<LoremIpsum />", () => {
  it("should render children correctly", () => {
    const content = "Lorem Ipsum";
    const { queryByText } = render(<LoremIpsum>{content}</LoremIpsum>);
    expect(queryByText("Lorem Ipsum")).toBeInTheDocument();
  });
});
'
`;
